src/solver/conv_asm_implicit_gemm_v4r1_dynamic.cpp

```diff
@@ -0,0 +1,589 @@
+                                                             ConstData_t bias_buf,
+                                                             const ConvolutionContext& ctx,
+                                                             const ConvSolution& solution,
+                                                             float& elapsed_time) const
+{
+    assert(bias_buf == nullptr);
+    (void)bias_buf;
+
+    return RunAndMeasureSolutionDynamicBase(
+        profile_h, bot_buf, top_buf, wei_buf, ctx, solution, elapsed_time);
+}
+
+int ConvAsmImplicitGemmV4R1DynamicFwd_1x1::RunAndMeasureSolution(miopen::Handle& profile_h,
+                                                                 ConstData_t bot_buf,
+                                                                 Data_t top_buf,
+                                                                 ConstData_t wei_buf,
+                                                                 ConstData_t bias_buf,
+                                                                 const ConvolutionContext& ctx,
+                                                                 const ConvSolution& solution,
+                                                                 float& elapsed_time) const
+{
+    assert(bias_buf == nullptr);
+    (void)bias_buf;
+
+    return RunAndMeasureSolutionDynamicBase(
+        profile_h, bot_buf, top_buf, wei_buf, ctx, solution, elapsed_time);
+}
+
+PerformanceImplicitGemmV4R1Dynamic
+ConvAsmImplicitGemmV4R1DynamicFwd::Search(const ConvolutionContext& context) const
+{
+    return GenericSearchFwd(*this, context);
+}
+
+PerformanceImplicitGemmV4R1Dynamic
+ConvAsmImplicitGemmV4R1DynamicFwd_1x1::Search(const ConvolutionContext& context) const
+{
+    return GenericSearchFwd(*this, context);
+}
+
+static inline ConvSolution GetSolutionBase(const ConvolutionContext& ctx,
+                                           const PerformanceImplicitGemmV4R1Dynamic& config,
+                                           const AsmImplicitGemmKernelV4R1Fwd_t& kernel_type)
+{
+    ConvSolution result;
+
+    std::string kernel_name = GetKernelNameImplicitGemmV4R1Dynamic(config, kernel_type);
+
+    int block_size = GetImplicitGemmV4R1DynamicBlockSize(config);
+    int grid_size  = GetImplicitGemmV4R1DynamicGridSize(ctx, config);
+
+    KernelInfo kernel;
+
+    kernel.kernel_file = "igemm_v4r1_dynamic.s";
+    kernel.kernel_name = kernel_name;
+    kernel.g_wk.clear();
+    /* Note here, for API like hipHccModuleLaunchKernel(), hipExtModuleLaunchKernel()
+    * grid dims is in unit of work item.
+    * But for api like hipModuleLaunchKernel(), grid dim is in unit of block.
+    */
+    kernel.g_wk.push_back(grid_size * block_size);
+    kernel.g_wk.push_back(1);
+    kernel.g_wk.push_back(1);
+    kernel.l_wk.clear();
+    kernel.l_wk.push_back(block_size);
+    kernel.l_wk.push_back(1);
+    kernel.l_wk.push_back(1);
+
+    kernel.comp_options = std::string(" -mcode-object-v3 ");
```

Comment:
Please look how other asm kernels denote CO v3 and fix this. Please make these solvers non-applicable for systems that do not support CO v3.

Reply:
By default ```-mcode-object-v3``` is included in hip_build_util.hpp for HCC
Any hip-clang doesn't need ```-mcode-object-v3```
Can we get rid of this flag from here? @atamazov

Reply:
No. But please see how other asm kernels do this.

Reply:
Code changed, and use `GenerateClangDefsym` to add a define for cov3 asm kernel.